src/com/ms/silverking/cloud/dht/crypto/SHA1Digest.java

```diff
@@ -4,25 +4,25 @@
 import java.security.NoSuchAlgorithmException;
 
 public class SHA1Digest {
-    private static ThreadLocal<MessageDigest>  tl = new ThreadLocal<MessageDigest>();
-    
-    public static final int BYTES = 20; 
-    
-    public static MessageDigest getLocalMessageDigest() {
-        MessageDigest   md;
-        
-        // FUTURE - ADD CODE FOR LWT THREADS TO GET THIS WITHOUT THE TL LOOKUP
-        // possibly a factory that we pass to LWT to generate our threads
-        md = tl.get();
-        if (md == null) {
-            try {
-                md = MessageDigest.getInstance("SHA-1");
-                tl.set(md);
-            } catch (NoSuchAlgorithmException nsae) {
-                throw new RuntimeException("panic");
-            }
-        }
-        md.reset();
-        return md;
+  private static ThreadLocal<MessageDigest> tl = new ThreadLocal<MessageDigest>();
+
+  public static final int BYTES = 20;
+
+  public static MessageDigest getLocalMessageDigest() {
+    MessageDigest md;
+
+    // FUTURE - ADD CODE FOR LWT THREADS TO GET THIS WITHOUT THE TL LOOKUP
+    // possibly a factory that we pass to LWT to generate our threads
+    md = tl.get();
+    if (md == null) {
+      try {
+        md = MessageDigest.getInstance("SHA-1");
+        tl.set(md);
+      } catch (NoSuchAlgorithmException nsae) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)